thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionTaskRunner.java

```diff
@@ -185,65 +193,57 @@ public AnomalyDetectionTaskRunner() {
     return results;
   }
 
+  @UnitOfWork
   private List<AnomalyResult> getExistingAnomalies() {
     List<AnomalyResult> results = new ArrayList<>();
 
     Session session = sessionFactory.openSession();
     try {
       ManagedSessionContext.bind(session);
-      Transaction transaction = session.beginTransaction();
-      try {
-        // The ones for this function
-        results.addAll(resultDAO.findAllByCollectionTimeAndFunction(collection, windowStart,
-            windowEnd, anomalyFunction.getSpec().getId()));
 
-        // The ones for any related functions
-        List<AnomalyFunctionRelation> relations =
-            relationDAO.findByParent(anomalyFunction.getSpec().getId());
-        for (AnomalyFunctionRelation relation : relations) {
-          results.addAll(resultDAO.findAllByCollectionTimeAndFunction(collection, windowStart,
-              windowEnd, relation.getChildId()));
-        }
+      // The ones for this function
+      results.addAll(resultDAO.findAllByCollectionTimeAndFunction(collection, windowStart,
+          windowEnd, anomalyFunction.getSpec().getId()));
 
-        transaction.commit();
-      } catch (Exception e) {
-        transaction.rollback();
-        e.printStackTrace();
+      // The ones for any related functions
+      List<AnomalyFunctionRelation> relations =
+          relationDAO.findByParent(anomalyFunction.getSpec().getId());
+      for (AnomalyFunctionRelation relation : relations) {
+        results.addAll(resultDAO.findAllByCollectionTimeAndFunction(collection, windowStart,
+            windowEnd, relation.getChildId()));
       }
+    } catch (Exception e) {
+      throw new RuntimeException(e);
     } finally {
       session.close();
       ManagedSessionContext.unbind(sessionFactory);
     }
-
     return results;
   }
 
+  @UnitOfWork
   private void handleResults(List<AnomalyResult> results) {
     Session session = sessionFactory.openSession();
```

Comment:
That doesn't seem to work "org.hibernate.HibernateException: No session currently bound to execution context" 
Remove the session handling and add it somewhere else?